pkg/stream/client.go

```diff
@@ -617,6 +617,17 @@ func (c *Client) BrokerForConsumer(stream string) (*Broker, error) {
 	if streamMetadata.responseCode != responseCodeOk {
 		return nil, lookErrorCode(streamMetadata.responseCode)
 	}
+
+	if streamMetadata.Leader == nil {
+		return nil, LeaderNotReady
+	}
+
+	for _, replica := range streamMetadata.Replicas {
+		if replica == nil {
+			return nil, ReplicaNotReady
```

Comment:
Maybe we should only filter the `nil` elements instead of a fatal error if any of them is not present? 

For example, if only the leader and 1 replica from a 3 member cluster are available, should we not connect to the cluster at all or should we just connect to one of the online members? I'd generally assume the latter, but I'm not sure if it works differently for streams.

So as a concrete idea, I'd suggest moving this loop to the point where we append to the final `brokers` slice (line 633 rn), and instead of returning an error we just skip (`continue`) the nil brokers, only appending the non-nil replicas one-by-one to the slice. WDYT?

Reply:
Not totally sure to understand. Can you please fork the repo e propose a PR to my branch ? With your suggestions ?
Thank you

Reply:
Sure! Will do tomorrow when I get back to the computer ok?
Sorry for the confusion

Reply:
No problem at all :)
Thank you for your contributions!

Reply:
Just created a PR, to merge on the branch of this current PR, with the suggested fix!

https://github.com/rabbitmq/rabbitmq-stream-go-client/pull/100